test/extensions/transport_sockets/tls/rsa_private_key_method_provider.cc

```diff
@@ -0,0 +1,258 @@
+#include "test/extensions/transport_sockets/tls/rsa_private_key_method_provider.h"
+
+#include <memory>
+
+#include "envoy/api/api.h"
+
+#include "openssl/ssl.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace PrivateKeyMethodProvider {
+
+int RsaPrivateKeyMethodProvider::ssl_rsa_connection_index = -1;
+
+void RsaPrivateKeyConnection::delayed_op() {
+  const std::chrono::milliseconds timeout_0ms{0};
+
+  timer_ = dispatcher_.createTimer([this]() -> void {
+    finished_ = true;
+    this->cb_.onPrivateKeyMethodComplete();
+    return;
+  });
+  timer_->enableTimer(timeout_0ms);
+}
+
+static ssl_private_key_result_t privateKeySign(SSL* ssl, uint8_t* out, size_t* out_len,
+                                               size_t max_out, uint16_t signature_algorithm,
+                                               const uint8_t* in, size_t in_len) {
+  RSA* rsa;
```

Comment:
nit: declare RSA and EVP_MD when it is needed and inline below

Reply:
Fixed.